common/src/deposits.rs

```diff
@@ -236,53 +375,53 @@ impl ReclaimScriptInputs {
-/// script does not start with <locked-time> OP_CHECKSEQUENCEVERIFY then
-/// we return an error.
-///
-/// laid out in https://github.com/stacks-network/sbtc/issues/30, where the
-/// script
-pub fn parse_reclaim_script(reclaim_script: &ScriptBuf) -> Result<ReclaimScriptInputs, Error> {
-    let (lock_time, script) = match reclaim_script.as_bytes() {
-        // These first two branches check for the case when the script is
-        // written with as few bytes as possible (called minimal CScriptNum
-        // format or something like that).
-        [0, OP_CSV, script @ ..] => (0, script),
-        // This catches numbers 1-16 and -1.
-        [n, OP_CSV, script @ ..]
-            if OP_PUSHNUM_NEG1 == *n || (OP_PUSHNUM_1..OP_PUSHNUM_16).contains(n) =>
-        {
-            (*n as i64 - OP_PUSHNUM_1 as i64 + 1, script)
-        }
-        // Numbers in bitcoin script are typically only 4 bytes (with a
-        // range from -2**31+1 to 2**31-1), unless we are working with
-        // OP_CSV or OP_CLTV, where 5-byte numbers are acceptable (with a
-        // range of -2**39+1 to 2**39-1).  See the following for how the
-        // code works in bitcoin-core:
-        // https://github.com/bitcoin/bitcoin/blob/v27.1/src/script/interpreter.cpp#L531-L573
-        [n, rest @ ..] if *n <= 5 && rest.get(*n as usize) == Some(&OP_CSV) => {
-            // We know the error and panic paths cannot happen because of
-            // the above `if` check.
-            let (script_num, [OP_CSV, script @ ..]) = rest.split_at(*n as usize) else {
-                return Err(Error::InvalidDepositScript);
-            };
-            (read_scriptint(script_num, 5)?, script)
-        }
-        _ => return Err(Error::InvalidReclaimScript),
-    };
+    /// Parse the reclaim script for the lock time.
+    ///
+    /// The goal of this function is to make sure that there are no
+    /// surprises in the reclaim script. These scripts are conceptually
+    /// very simple and are their format is
+    /// ```text
+    ///  <locked-time> OP_CHECKSEQUENCEVERIFY <rest-of-reclaim-script>
+    /// ```
+    /// This function extracts the <locked-time> from the script. If the
+    /// script does not start with <locked-time> OP_CHECKSEQUENCEVERIFY
+    /// then we return an error.
+    ///
+    /// laid out in https://github.com/stacks-network/sbtc/issues/30, where
+    /// the script
+    pub fn parse(reclaim_script: &ScriptBuf) -> Result<Self, Error> {
+        let (lock_time, script) = match reclaim_script.as_bytes() {
+            // These first two branches check for the case when the script
+            // is written with as few bytes as possible (called minimal
+            // CScriptNum format or something like that).
+            [0, OP_CSV, script @ ..] => (0, script),
+            // This catches numbers 1-16 and -1.
+            [n, OP_CSV, script @ ..]
+                if OP_PUSHNUM_NEG1 == *n || (OP_PUSHNUM_1..OP_PUSHNUM_16).contains(n) =>
```

Comment:
Per [BIP112 Summary](https://github.com/bitcoin/bips/blob/master/bip-0112.mediawiki#summary), it says the script will terminate if the top item of the stack is < 0. Am I reading that wrong and that -1 is valid?

Reply:
No, you read it correctly. I reject it later, but because I thought that it would be non-standard, I forgot that it was invalid too. I'll reject it here as well.

Reply:
Oh, I should just note that negative number are invalid. It's probably best to catch this all in one place.

Reply:
Done in https://github.com/stacks-network/sbtc/pull/350/commits/63b46c04ca17e1f877a55851e0b9d97b3b9bf970